### PR TITLE
[IMPROVEMENT] Add Setup For Initiating Python 3.7 Compatibility

### PR DESCRIPTION
**Pull request by thealphadollar**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [ ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [x] I am an active contributor to the project.

---

The PR concerns #281 and brings in the following changes.

- Update python version to 3.7 in Pipfile
- Add MonkeyType to Pipfile
- Add PyType to Pipfile
- Skip E701 from Pep8 since we will be using the syntax `x: int = 9`
- Refactor README.md with better developer instructions including using PyType and MonkeyType
- Add annotations to all root modules.
- Make Travis Builds on 3.4, 3.5, 3.6 optional to pass since we are switching to 3.7
- Move Travis build dist from trust (default) to xenial since Python 3.7 is not available on trust.
- Move PEP8 checking out of unit-tests
- Move travis installed test requirements to test-requirements.txt

After this I'll be adding unit-tests first and then only adding more annotations.

NOTE: For reasoning on each change, please see the message for the commit that introduces the change.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,32 @@
+dist: xenial
 language: python
 python:
   - 3.4
```
Comment (canihavesomecoffee) [inline .travis.yml]: If we bump the minimal version to 3.7, we should drop testing older versions ;)

File: .travis.yml [hunk 2]
```diff
@@ -1,36 +1,32 @@
...
+  - pip install -r test-requirements.txt
   - pip install -r requirements.txt
-  - pip install Flask-Testing
-  - pip install blinker
 script:
   - dodgy
-  - mypy mod_*
-  - pydocstyle --config=./.pydocstylerc
   - isort -rc --check-only
+  - pydocstyle --config=./.pydocstylerc
```
Comment (canihavesomecoffee) [inline .travis.yml]: Why call this twice with minimal changes in how it's called?

File: .travis.yml [hunk 3]
```diff
@@ -1,36 +1,32 @@
+dist: xenial
 language: python
 python:
   - 3.4
   - 3.5
   - 3.6
-  - nightly
```
Comment (aadibajpai) [inline .travis.yml]: why not keep the nightly too given 3.8 beta is already out

File: database.py [hunk 4]
```diff
@@ -1,11 +1,15 @@
 """handles database session and data-type across app."""
+from __future__ import annotations
```
Comment (canihavesomecoffee) [inline database.py]: I see no use of this import? Can you point out where? PRobably overlooking it.

**Comment by thealphadollar** [comment]
Yeah, thanks for the note.

I'll add it as an optional check.

As discussed with @canihavesomecoffee, we will be focusing mainly on 3.7
and hence only that build will be mandatory to pass.

File: .travis.yml (hunk 2 above)
Comment (thealphadollar) [reply .travis.yml]: One is py**doc**style and the other is py**code**style

File: database.py (hunk 4 above)
Comment (thealphadollar) [reply database.py]: https://www.python.org/dev/peps/pep-0563/#id6

This is used to use classes which have not been completely defined but we still want to use them in annotations.

"DeclEnum" L96 function definition is an example of the same.

**Pull request merged**